inference-engine/thirdparty/ocv/opencv_hal_neon.hpp

```diff
@@ -2323,7 +2323,131 @@ inline v_uint16x8 v_mulhi(const v_uint16x8& a, uint16_t b) {
+    result = vsetq_lane_s32(*reinterpret_cast<const int*>(&tmp[4 * (chanNum * (*(index + shift + 1)) + c)]), result, 1);
+    result = vsetq_lane_s32(*reinterpret_cast<const int*>(&tmp[4 * (chanNum * (*(index + shift + 0) + 1) + c)]), result, 2);
+    result = vsetq_lane_s32(*reinterpret_cast<const int*>(&tmp[4 * (chanNum * (*(index + shift + 1) + 1) + c)]), result, 3);
+
+    vec.val = vreinterpretq_u8_s32(result);
+}
+
+namespace {
+template<int chanNum>
+CV_ALWAYS_INLINE v_int16x8 v_gather_channel(const uchar src[], const short* index, int channel, int pos) {
+    int16x8_t result;
+    result = vsetq_lane_s16(*reinterpret_cast<const uchar*>(&src[chanNum * (*index + pos) + channel]), result, 0);
+    result = vsetq_lane_s16(*reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 1) + pos) + channel]), result, 1);
+    result = vsetq_lane_s16(*reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 2) + pos) + channel]), result, 2);
+    result = vsetq_lane_s16(*reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 3) + pos) + channel]), result, 3);
+    result = vsetq_lane_s16(*reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 4) + pos) + channel]), result, 4);
+    result = vsetq_lane_s16(*reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 5) + pos) + channel]), result, 5);
+    result = vsetq_lane_s16(*reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 6) + pos) + channel]), result, 6);
+    result = vsetq_lane_s16(*reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 7) + pos) + channel]), result, 7);
+    return v_int16x8(result);
+}
+}  // namespace
+
+template<int imm>
+CV_ALWAYS_INLINE v_uint8x16 v_blend(const v_uint8x16& a, const v_uint8x16& b)
+{
+    const uint16_t _mask[8] = { ((imm) & (1 << 0)) ? 0xFFFF : 0x0000,
+                                ((imm) & (1 << 1)) ? 0xFFFF : 0x0000,
+                                ((imm) & (1 << 2)) ? 0xFFFF : 0x0000,
+                                ((imm) & (1 << 3)) ? 0xFFFF : 0x0000,
+                                ((imm) & (1 << 4)) ? 0xFFFF : 0x0000,
+                                ((imm) & (1 << 5)) ? 0xFFFF : 0x0000,
+                                ((imm) & (1 << 6)) ? 0xFFFF : 0x0000,
+                                ((imm) & (1 << 7)) ? 0xFFFF : 0x0000 };
+    uint16x8_t _mask_vec = vld1q_u16(_mask);
+    uint16x8_t _a = vreinterpretq_u16_u8(a.val);
+    uint16x8_t _b = vreinterpretq_u16_u8(b.val);
+    return v_uint8x16(vreinterpretq_u8_u16(vbslq_u16(_mask_vec, _b, _a)));
+}
+
+CV_ALWAYS_INLINE v_uint8x16 v_shuffle(const v_uint8x16& a, const v_uint8x16& mask)
+{
+
+    uint8x16_t tbl = a.val;   // input a
+    uint8x16_t idx = mask.val;  // input mask
+    uint8x16_t idx_masked =
+               vandq_u8(idx, vdupq_n_u8(0x8F));  // avoid using meaningless bits
+#if defined(__aarch64__)
+    return v_uint8x16(vqtbl1q_u8(tbl, idx_masked));
```

Comment:
for 32-bit ARM architecture it will not work?

Reply:
This instruction is supported with AARCH64 only.